src/tasks/move_directory.rs

```diff
@@ -0,0 +1,45 @@
+use std::{fs, path};
+use tracing_batteries::prelude::tracing;
+use crate::core::{Core, Target};
+use crate::{errors};
+use crate::tasks::{Task};
+
+pub struct MoveDirectory {
+    pub new_path: path::PathBuf,
+}
+
+#[async_trait::async_trait]
+impl Task for MoveDirectory {
+    #[tracing::instrument(name = "task:git_clone(repo)", err, skip(self, _core))]
+    async fn apply_repo(&self, _core: &Core, repo: &crate::core::Repo) -> Result<(), crate::core::Error> {
+        if !repo.exists() {
+            return Err(errors::user(
+                "Repository does not exist.",
+                "Provide the name of the repository you wish to rename.",
+            ))
```

Comment:
Minor thing, but since this error message is reasonably going to be presented to a user, it might be nice to include a bit more detail:

```rust
errors::user(
  format!("The repository {} does not exist on your machine and cannot be moved as a result.", repo.full_name), 
  format!("Make sure the name is correct and that the repository exists by running `git-tool clone {}` first.", repo.full_name)
)
```

Reply:
yup. Fully aligned